kornia/enhance/jpeg.py

```diff
@@ -0,0 +1,566 @@
+    output_c: Tensor = rescale(
+        input_c[:, None], factor=2.0, interpolation="bilinear", align_corners=False, antialias=False
+    )
+    return output_c[:, 0]
+
+
+def _jpeg_encode(
+    image_rgb: Tensor,
+    jpeg_quality: Tensor,
+    quantization_table_y: Tensor,
+    quantization_table_c: Tensor,
+) -> Tuple[Tensor, Tensor, Tensor]:
+    """Performs JPEG encoding.
+
+    Args:
+        image_rgb (Tensor): RGB input images of the shape :math:`(B, 3, H, W)`.
+        jpeg_quality (Tensor): Compression strength of the shape :math:`(B)`.
+        quantization_table_y (Tensor): Quantization table for Y channel.
+        quantization_table_c (Tensor): Quantization table for C channels.
+
+    Returns:
+        y_encoded (Tensor): Encoded Y component of the shape :math:`(B, N, 8, 8)`.
+        cb_encoded (Tensor): Encoded Cb component of the shape :math:`(B, N, 8, 8)`.
+        cr_encoded (Tensor): Encoded Cr component of the shape :math:`(B, N, 8, 8)`.
+    """
+    # Convert RGB image to YCbCr.
+    image_ycbcr: Tensor = rgb_to_ycbcr(image_rgb)
+    # Scale pixel-range to [0, 255]
+    image_ycbcr = 255.0 * image_ycbcr
+    # Perform chroma subsampling
+    input_y, input_cb, input_cr = _chroma_subsampling(image_ycbcr)
+    # Patchify, DCT, and rounding
+    input_y, input_cb, input_cr = _patchify_8x8(input_y), _patchify_8x8(input_cb), _patchify_8x8(input_cr)
+    dct_y, dct_cb, dct_cr = _dct_8x8(input_y), _dct_8x8(input_cb), _dct_8x8(input_cr)
+    y_encoded: Tensor = _quantize(
+        dct_y,
+        jpeg_quality,
+        quantization_table_y,
+    )
+    cb_encoded: Tensor = _quantize(
+        dct_cb,
+        jpeg_quality,
+        quantization_table_c,
+    )
+    cr_encoded: Tensor = _quantize(
+        dct_cr,
+        jpeg_quality,
+        quantization_table_c,
+    )
+    return y_encoded, cb_encoded, cr_encoded
+
+
+def _jpeg_decode(
+    input_y: Tensor,
+    input_cb: Tensor,
+    input_cr: Tensor,
+    jpeg_quality: Tensor,
+    H: int,
+    W: int,
+    quantization_table_y: Tensor,
+    quantization_table_c: Tensor,
+) -> Tensor:
+    """Performs JPEG decoding.
+
+    Args:
+        input_y (Tensor): Compressed Y component of the shape :math:`(B, N, 8, 8)`.
+        input_cb (Tensor): Compressed Cb component of the shape :math:`(B, N, 8, 8)`.
+        input_cr (Tensor): Compressed Cr component of the shape :math:`(B, N, 8, 8)`.
+        jpeg_quality (Tensor): Compression strength of the shape :math:`(B)`.
+        H (int): Original image height.
+        W (int): Original image width.
+        quantization_table_y (Tensor): Quantization table for Y channel.
+        quantization_table_c (Tensor): Quantization table for C channels.
+
+    Returns:
+        rgb_decoded (Tensor): Decompressed RGB image of the shape :math:`(B, 3, H, W)`.
+    """
+    # Dequantize inputs
+    input_y = _dequantize(
+        input_y,
+        jpeg_quality,
+        quantization_table_y,
+    )
+    input_cb = _dequantize(
```

Comment:
stupid question, why you cannot vectorise the process for the cb/cr channels in a single operation?

Reply:
Done in both the encoding and decoding.

Reply:
Very valid question btw ;)